File: internal/templates/templates/todo-api/_node/services/delete-todo/handler.mjs
// DELETE /todos/{id} — remove one todo.
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, DeleteCommand } from "@aws-sdk/lib-dynamodb";

const db = DynamoDBDocumentClient.from(new DynamoDBClient({}));

export const handler = async (event) => {
  // Deleting something absent is fine — DELETE is idempotent in REST.
  await db.send(new DeleteCommand({
    TableName: process.env.TABLE_NAME,
    Key: { id: event.pathParameters.id },
  }));
  return { statusCode: 204, body: "" };
};

File: internal/templates/templates/todo-api/_python/services/delete-todo/handler.py
"""DELETE /todos/{id} — remove one todo."""

import os

import boto3

table = boto3.resource("dynamodb").Table(os.environ["TABLE_NAME"])


def handler(event, context):
    # Deleting something absent is fine — DELETE is idempotent in REST.
    table.delete_item(Key={"id": event["pathParameters"]["id"]})
    return {"statusCode": 204, "body": ""}
